lib/vocabularies/applicator/patternProperties.ts

```diff
@@ -12,9 +13,16 @@ const def: CodeKeywordDefinition = {
   code(cxt: KeywordCxt) {
     const {gen, schema, data, parentSchema, it} = cxt
     const {opts} = it
-    const patterns = schemaProperties(it, schema)
-    // TODO mark properties matching patterns with always valid schemas as evaluated
-    if (patterns.length === 0) return
+    const patterns = allSchemaProperties(schema)
+    const validPatterns = patterns.filter((p) => !alwaysValidSchema(it, schema[p] as AnySchema))
+    const checkForUnevaluatedProperties =
+      parentSchema.unevaluatedProperties !== true &&
+      parentSchema.unevaluatedProperties !== undefined
```

Comment:
I don't think you can limit it to checking parent - we should mark unevaluatedProperties regardless - it can be higher up in the schema...